carina-utils/src/main/java/com/qaprosoft/carina/core/foundation/utils/tag/TagManager.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright 2013-2018 QaProSoft (http://www.qaprosoft.com).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.qaprosoft.carina.core.foundation.utils.tag;
+
+import org.apache.log4j.Logger;
+import org.testng.ITestResult;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+
+public class TagManager {
+    protected static final Logger LOGGER = Logger.getLogger(TagManager.class);
+
+    private TagManager() {
+    }
+
+    public static HashMap<String, String> getTag(ITestResult result) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The return type of this method should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://52.53.251.139/sonarqube/coding_rules#rule_key=squid%3AS1319)